src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > elementData.length) {
+            throw new ArrayIndexOutOfBoundsException("Your ArrayIndex are out of bounds");
+        } else {
+            if (size == elementData.length) {
+                grow();
+            } else {
+                System.arraycopy(elementData, index, elementData, index + 1, size - index);
+                elementData[index] = value;
+                size++;
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
+    //// как создать проверку на не существующий объект
     @Override
     public T get(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException("Your ArrayIndex are out of bounds");
+        }
+        for (int i = 0; i < elementData.length; i++) {
+            if (elementData[i] == elementData[index]) {
+                return elementData[i];
+            }
+        }
         return null;
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException("Your ArrayIndex are out of bounds");
+        }
+        for (int i = 0; i < elementData.length; i++) {
+            if (elementData[i] == elementData[index]) {
+                elementData[index] = value;
+            }
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T temp;
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException("Your ArrayIndex are out of bounds");
+        }
+        temp = elementData[index];
+        elementData[index] = null;
```

Comment:
```suggestion
        T temp = elementData[index];
```
we don't need line 82 - we will shift values to the left on the next line, so the value will be changed

Reply:
I understand. done